bindings/time-profiler.cc

```diff
@@ -19,6 +19,13 @@
 
 using namespace v8;
 
+#if (NODE_MODULE_VERSION > NODE_8_0_MODULE_VERSION)
+CpuProfiler* cpuProfiler = CpuProfiler::New(v8::Isolate::GetCurrent());
```

Comment:
I feel you maybe got a memory leak. Don't you need to dispose the profiler with the new flow? The name change from Get() -> New() implies some changes in how the life time is managed I think. I also think you want to retain it between the profiling sessions to avoid re-creating it.

Reply:
The variable is at file global scope, so the profiler is being retained across the profiling sessions. The lack of a [`CpuProfiler::Dispose`](https://cs.chromium.org/chromium/src/v8/include/v8-profiler.h?type=cs&sq=package:chromium&g=0&l=303) may seem inelegant, but I wouldn't call it a leak. The profiler exists for the lifetime of the application.

Reply:
The question I have is what happens if there are multiple modules in the system that create the CPU profiler using `CpuProfiler::New`. Would both of them end up working correctly? /cc @a1ph

Reply:
Apologies, I missed that it's in global scope. I'd add a comment that the leak is intended.

Reply:
By the way, completely aside but: we ignore the possibility of existence of multiple isolates, right?

Reply:
You can create as many profilers as you need. You can even have several for the same isolate. We have a test for that https://cs.chromium.org/chromium/src/v8/test/cctest/test-cpu-profiler.cc?type=cs&q=MultipleProfilers&sq=package:chromium&g=0&l=2472

Reply:
Added a comment stating that `Dispose` is not called intentionally.